src/Microsoft.AspNetCore.Hosting/Internal/ActivityTrackingOptions.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    public class ActivityTrackingOptions
```

Comment:
I would strongly suggest following Christian's suggestion.    It is HARD to get extensibility correct, and any PUBLIC surface has a compatibility requirement.  Thus it has the potential to be 'instant legacy' (that is no one uses it, but you have to support it forever).    We have learned the hard wary, that generally speaking if you don't have a reason to make something public, DON'T.    Thus unless you have people who ACTUALLY need this extensibility (and this is ALL they need to make their scenario work), you should not expose it at this time.